src/mscorlib/shared/System/Globalization/GregorianCalendar.cs

```diff
@@ -382,6 +382,22 @@ public override int GetYear(DateTime time)
             return time.Year;
         }
 
+        internal override bool IsValidYear(int year, int era) => year >= 1 && year <= MaxYear;
+
+        internal override bool IsValidDay(int year, int month, int day, int era)
```

Comment:
>internal override bool IsValidDay(int year, int month, int day, int era) [](start = 8, length = 72)

I am seeing in other calendars, this can throw but we are not throwing here. is it in purpose?

Reply:
> is it in purpose?

It doesn't throw with the base implementation, does it?  I don't want to introduce exceptions where they didn't exist before.

Reply:
in the base Calendar.cs 

http://source.dot.net/#System.Private.CoreLib/shared/System/Globalization/Calendar.cs,769

IsValidDay is calling GetDaysInMonth which can throw

http://source.dot.net/#System.Private.CoreLib/shared/System/Globalization/GregorianCalendar.cs,291

Reply:
Yes, but it's calling into that after the arguments that are passed in have already been verified, no?  The only one I see that isn't validated is era.  Is there a way an era other than CurrentEra or ADEra could make it into IsValidDay?

Reply:
> Is there a way an era other than CurrentEra or ADEra could make it into IsValidDay?

I don't think so.

Reply:
> I don't think so.

Ok.  In that case, do you see a way that the existing implementation could actually throw?  I can certainly add throwing checks if it's needed, I just don't currently see how they would ever be hit.  I could be wrong, though.

Reply:
I think your implementation will work. I have reviewed the original code carefully and I am not seeing a way we can end up throwing (at least for the Gregorian calendar). so I am fine with your changes here.

Reply:
Thanks for verifying, @tarekgh.